[BUG] Panic on rebuild when using watch --no-up

### Description

Using a 'rebuild' action with the 'watch --no-up' method results in a go panic. It works when using 'up --watch', and the 'sync' action works with both.

A minimal reproduction repo can be found here: 
https://github.com/norbert-balazs/docker-bug-repro

### Steps To Reproduce

1. Check out repro repo
2. `docker compose up --build -d` (bug also appears if '-d' is not used)
3. `docker compose watch --no-up`
4. Write something to watched-file.txt
5. Observe crash of `watch --no-up`

### Compose Version

```Text
Docker Compose version v5.0.2
```

### Docker Environment

```Text
Client: Docker Engine - Community
 Version:    29.2.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.31.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v5.0.2
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 3
  Running: 0
  Paused: 0
  Stopped: 3
 Images: 57
 Server Version: 29.2.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 CDI spec directories:
  /etc/cdi
  /var/run/cdi
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: dea7da592f5d1d2b7755e3a161be07f43fad8f75
 runc version: v1.3.4-0-gd6d73eb8
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.12.69+deb12-amd64
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: x86_64
 CPUs: 22
 Total Memory: 62.24GiB
 Name: [REDACTED] 
 ID: 3bae2169-a47f-48c0-8399-0a90b2cb2036
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  ::1/128
  127.0.0.0/8
 Live Restore Enabled: false
 Firewall Backend: iptables+firewalld
```

### Anything else?

Error on rebuild:
```
Watch enabled
Syncing service "something" after 1 changes were detected
Rebuilding service(s) ["something"] after changes were detected...
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xf9540b]

goroutine 81 [running]:
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End.deferwrap1()
        go.opentelemetry.io/otel/sdk@v1.38.0/trace/span.go:468 +0x25
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End(0xc00001a3c0, {0x0, 0x0, 0xc000309c00?})
        go.opentelemetry.io/otel/sdk@v1.38.0/trace/span.go:517 +0xc12
panic({0x1197780?, 0x1e7da50?})
        runtime/panic.go:792 +0x132
time.(*Ticker).Stop(...)
        time/tick.go:53
github.com/docker/compose/v5/cmd/display.(*ttyWriter).event(0xc0004601b0, {{0xc00053f140, 0x13}, {0x0, 0x0}, {0x1370a35, 0x8}, {0x0, 0x0}, 0x0, ...})
        github.com/docker/compose/v5/cmd/display/tty.go:204 +0x4b
github.com/docker/compose/v5/cmd/display.(*ttyWriter).On(0xc0004601b0, {0xc0001245a0?, 0xc00029ea30?, 0xc0005254a0?})
        github.com/docker/compose/v5/cmd/display/tty.go:197 +0x245
github.com/docker/compose/v5/pkg/compose.(*composeService).doBuildBake(0xc000530000, {0x1580968, 0xc000394960}, 0xc0002e4fa0, 0xc0003949c0, {0x0, 0x0, {0x136c540, 0x5}, 0xc000379dd0, ...})
        github.com/docker/compose/v5/pkg/compose/build_bake.go:209 +0xdb8
github.com/docker/compose/v5/pkg/compose.(*composeService).build(0xc000530000, {0x1580968, 0xc000394960}, 0xc0003baf00, {0x0, 0x0, {0x136c540, 0x5}, 0xc000379dd0, 0x0, ...}, ...)
        github.com/docker/compose/v5/pkg/compose/build.go:105 +0x3c6
github.com/docker/compose/v5/pkg/compose.(*composeService).rebuild.func2({0x1580968?, 0xc000394960?})
        github.com/docker/compose/v5/pkg/compose/watch.go:645 +0xa6
github.com/docker/compose/v5/pkg/compose.(*composeService).rebuild.SpanWrapFunc.func3({0x15809a0, 0xc000408190})
        github.com/docker/compose/v5/internal/tracing/wrap.go:43 +0x13d
github.com/docker/compose/v5/pkg/compose.(*composeService).rebuild(0xc000530000, {0x15809a0, 0xc000408190}, 0xc0003baf00, {0xc000334550, 0x1, 0x1}, {0xc000729340, {0x157df10, 0xc0002ce000}, ...})
        github.com/docker/compose/v5/pkg/compose/watch.go:647 +0x404
github.com/docker/compose/v5/pkg/compose.(*composeService).handleWatchBatch(0xc000530000, {0x15809a0, 0xc000408190}, 0xc0003baf00, {0xc000729340, {0x157df10, 0xc0002ce000}, 0x1, {0xc000503280, 0x0, ...}}, ...)
        github.com/docker/compose/v5/pkg/compose/watch.go:564 +0x3fa
github.com/docker/compose/v5/pkg/compose.(*composeService).watchEvents(0xc000530000, {0x15809a0?, 0xc0004080f0?}, 0xc0003baf00, {0xc000729340, {0x157df10, 0xc0002ce000}, 0x1, {0xc000503280, 0x0, ...}}, ...)
        github.com/docker/compose/v5/pkg/compose/watch.go:382 +0x465
github.com/docker/compose/v5/pkg/compose.(*composeService).watch.func1()
        github.com/docker/compose/v5/pkg/compose/watch.go:283 +0x86
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.19.0/errgroup/errgroup.go:93 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        golang.org/x/sync@v0.19.0/errgroup/errgroup.go:78 +0x93
```

--- a/cmd/display/tty.go
+++ b/cmd/display/tty.go
@@ -178,7 +178,9 @@ func (w *ttyWriter) Done(operation string, success bool) {
 	w.print()
 	w.mtx.Lock()
 	defer w.mtx.Unlock()
-	w.ticker.Stop()
+	if w.ticker != nil {
+		w.ticker.Stop()
+	}
 	w.operation = ""
 	w.done <- true
 }
@@ -202,12 +204,14 @@ func (w *ttyWriter) On(events ...api.Resource) {
 
 func (w *ttyWriter) event(e api.Resource) {
 	// Suspend print while a build is in progress, to avoid collision with buildkit Display
-	if e.Text == api.StatusBuilding {
-		w.ticker.Stop()
-		w.suspended = true
-	} else if w.suspended {
-		w.ticker.Reset(100 * time.Millisecond)
-		w.suspended = false
+	if w.ticker != nil {
+		if e.Text == api.StatusBuilding {
+			w.ticker.Stop()
+			w.suspended = true
+		} else if w.suspended {
+			w.ticker.Reset(100 * time.Millisecond)
+			w.suspended = false
+		}
 	}
 
 	if last, ok := w.tasks[e.ID]; ok {

